executor/simple.go

```diff
@@ -659,6 +653,12 @@ func (e *SimpleExec) executeCreateUser(ctx context.Context, s *ast.CreateUserStm
 		if s.IsCreateRole {
 			user = fmt.Sprintf(`('%s', '%s', '%s', 'Y')`, spec.User.Hostname, spec.User.Username, pwd)
 		}
+		if exists {
+			if !s.IfNotExists {
```

Comment:
I found MySQL used a different error type:

```
mysql> CREATE USER 'newuser'@'localhost';
Query OK, 0 rows affected (0.00 sec)

mysql> CREATE USER 'newuser'@'localhost';
ERROR 1396 (HY000): Operation CREATE USER failed for 'newuser'@'localhost'
```
PTAL, however, I think using `ErrUserAlreadyExists` is better.